### PR TITLE
Activate Prover node

### PR DESCRIPTION
**Pull request by eyusufatik**
# Description
Activates prover node

### CODE REVIEW

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs [hunk 1]
```diff
@@ -229,341 +225,350 @@ where
...
+                });
+
+            self.ledger_db
+                .set_prover_last_scanned_l1_height(SlotNumber(l1_height))
+                .unwrap_or_else(|_| {
+                    panic!(
+                        "Failed to put prover last scanned l1 height in the ledger db {}",
+                        l1_height
+                    )
+                });
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/runner.rs:562]: return Err

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/mod.rs [hunk 2]
```diff
@@ -125,7 +134,11 @@ where
             <Self::DaService as DaService>::Spec,
         >,
     ) -> WitnessSubmissionStatus {
-        self.prover_state.submit_witness(state_transition_data)
+        let status = self.prover_state.submit_witness(state_transition_data);
+
+        tracing::info!("Witness submission status: {:?}", status);
+
+        status
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/mod.rs:141]: revert

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/mod.rs [hunk 3]
```diff
@@ -143,11 +157,37 @@ where
...
+                            da_data.try_to_vec().expect("Should serialize").as_slice(),
+                        )
+                        .await
+                        .map_err(|e| anyhow::anyhow!(e))?;
+
+                    break Ok((tx_id, proof));
+                }
+                ProverStatus::ProvingInProgress => {
+                    tracing::info!("Proof generation is in progress");
+                    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/mod.rs:185]: better if 5 secs

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs [hunk 4]
```diff
@@ -149,15 +142,17 @@ where
 
         match prover_status {
             ProverStatus::WitnessSubmitted(state_transition_data) => {
+                tracing::info!("here");
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs:145]: revert

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs [hunk 5]
```diff
@@ -149,15 +142,17 @@ where
...
                 let start_prover = prover_state.inc_task_count_if_not_busy(self.num_threads);
-
+                tracing::info!(start_prover);
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs:147]: revert

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs [hunk 6]
```diff
@@ -149,15 +142,17 @@ where
...
                 // Initiate a new proving job only if the prover is not busy.
                 if start_prover {
                     prover_state.set_to_proving(block_header_hash.clone());
                     vm.add_hint(state_transition_data);
-
+                    tracing::info!("added hint");
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs:152]: revert

File: crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs [hunk 7]
```diff
@@ -149,15 +142,17 @@ where
...
                     self.pool.spawn(move || {
                         tracing::info_span!("guest_execution").in_scope(|| {
+                            tracing::info!("going into make proof");
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-stf-runner/src/prover_service/parallel/prover.rs:155]: Starting proving

File: crates/sovereign-sdk/full-node/sov-ledger-rpc/src/server.rs [hunk 8]
```diff
@@ -204,6 +207,14 @@ where
         },
     )?;
 
+    rpc.register_async_method("ledger_getProofBySlotHeight", |params, ledger| async move {
+        // Returns proof on DA slot with given height
+        let height: u64 = params.one()?;
+        ledger
+            .get_proof_data_by_l1_height(height)
+            .map_err(|e| to_jsonrpsee_error_object(LEDGER_RPC_ERROR, e))
+    })?;
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/sov-ledger-rpc/src/server.rs:216]: getProofBySlotHash

File: crates/sovereign-sdk/full-node/db/sov-db/src/schema/types.rs [hunk 9]
```diff
@@ -70,6 +72,62 @@ pub struct StoredSlot {
...
+        initial_state_root: stored_state_transition.initial_state_root,
+        final_state_root: stored_state_transition.final_state_root,
+        state_diff: stored_state_transition.state_diff,
+        da_slot_hash: stored_state_transition.da_slot_hash,
+        sequencer_da_public_key: stored_state_transition.sequencer_da_public_key,
+        sequencer_public_key: stored_state_transition.sequencer_public_key,
+        validity_condition: stored_state_transition.validity_condition,
+    }
+}
+
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/db/sov-db/src/schema/types.rs:122]: can be a from implementation

File: crates/sovereign-sdk/full-node/db/sov-db/src/schema/types.rs [hunk 10]
```diff
@@ -70,6 +72,62 @@ pub struct StoredSlot {
...
+/// Converts proof data to hex encoded rpc response
+pub fn convert_to_rpc_proof(stored_proof: Proof) -> ProofRpcResponse {
+    match stored_proof {
+        Proof::Full(data) => ProofRpcResponse::Full(data),
+        Proof::PublicInput(data) => ProofRpcResponse::PublicInput(data),
+    }
+}
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/db/sov-db/src/schema/types.rs:129]: same

File: crates/sovereign-sdk/full-node/db/sov-db/src/schema/types.rs [hunk 11]
```diff
@@ -70,6 +72,62 @@ pub struct StoredSlot {
...
+#[derive(Debug, PartialEq, BorshDeserialize, BorshSerialize)]
+pub struct StoredProof {
+    /// Tx id
+    pub l1_tx_id: [u8; 32],
+    /// Proof
+    pub proof: Proof,
+    /// State transition
+    pub state_transition: StoredStateTransition,
+}
+
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/db/sov-db/src/schema/types.rs:85]: single from impl. will remove those convert functions

File: bin/citrea/provers/risc0/guest-mock/Cargo.lock [hunk 12]
```diff
@@ -2904,6 +2904,7 @@ dependencies = [
  "sov-modules-api",
  "sov-modules-stf-blueprint",
  "sov-rollup-interface",
+ "tracing",
```
[the remark this answers is not in the document]
Comment (ercecan) [reply bin/citrea/provers/risc0/guest-mock/Cargo.lock:2907]: this is coming from sov-stf-runner so it is ok

[the remark this answers is not in the document]
Comment (ercecan) [reply bin/citrea/provers/risc0/guest-mock/Cargo.lock:2907]: nvm removed

**Review by ercecan — APPROVED** [review]

**Pull request merged**